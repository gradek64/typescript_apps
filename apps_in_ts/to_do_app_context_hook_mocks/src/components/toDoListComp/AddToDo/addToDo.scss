@import '../../../variables';

.formTask {
  width: 90%;
  margin: 0 auto;
  display: flex;

  input.text {
    flex-grow: 1;
    border-color: transparent;
    border-radius: 10px 0 0 10px;
    font-size: 14px;
    padding: 5px;
  }

  input[type='submit'] {
    background-color: #bebed7;
    border-color: transparent;
    border-radius: 0 10px 10px 0;
    padding: 5px;
  }
}

@media only screen and (max-width: $mobile) {
  .formTask {
    flex-direction: column;
    justify-content: center;
    align-items: center;

    input.text,
    input[type='submit'] {
      border-radius: 10px;
      width: 100%;
      margin: 5px;
    }
  }
}
