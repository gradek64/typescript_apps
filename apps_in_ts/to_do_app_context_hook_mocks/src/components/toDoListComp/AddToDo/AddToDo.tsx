import React, { useState } from 'react';
import { useTasks } from '../../Context/TaskProvider';
import './addToDo.scss';

const NewTaskForm = () => {
  const [task, setTask] = useState('');
  const { tasks, newTask } = useTasks();

  const submit = (e: any) => {
    e.preventDefault();
    if (typeof newTask === 'function')
      newTask({ todo: task, complete: false, id: `task${tasks.length + 1}` });
  };

  return (
    <form onSubmit={submit} className="formTask">
      <input
        type="text"
        value={task}
        className="text"
        onChange={(e) => {
          setTask(e.target.value);
        }}
        placeholder="Get some eggs..."
        required
      />
      <input type="submit" value="ADD" />
    </form>
  );
};
export default NewTaskForm;
