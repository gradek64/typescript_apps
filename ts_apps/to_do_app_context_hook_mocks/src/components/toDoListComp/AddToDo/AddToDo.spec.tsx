import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import AddToDo from './AddToDo';

describe('should render correclty with initial props', () => {
  afterEach(cleanup);
  it('should render and take a snapshot', () => {
    const { asFragment } = render(<AddToDo />);
    const component = asFragment();
    expect(component).toMatchSnapshot();
  });
});

describe('events', () => {
  it('submits the form with button click', () => {
    const { container } = render(<AddToDo />);
    const form = container.querySelectorAll('form')[0];
    form.submit = jest.fn();
    const bttn = container.querySelectorAll('input')[1];
    fireEvent.click(bttn);
    setTimeout(() => {
      expect(form.submit).toBeCalled();
    }, 0);
  });
});
