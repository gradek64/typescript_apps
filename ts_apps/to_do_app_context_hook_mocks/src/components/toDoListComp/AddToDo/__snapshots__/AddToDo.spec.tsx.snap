// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correclty with initial props should render and take a snapshot 1`] = `
<DocumentFragment>
  <form
    class="formTask"
  >
    <input
      class="text"
      placeholder="Get some eggs..."
      required=""
      type="text"
      value=""
    />
    <input
      type="submit"
      value="ADD"
    />
  </form>
</DocumentFragment>
`;
